webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTable.js

```diff
@@ -8,42 +8,40 @@ import { translate as __ } from '../../common/I18n';
 import createModelsTableSchema from './ModelsTableSchema';
 import { getURIQuery } from '../../common/helpers';
 
-class ModelsTable extends React.Component {
-  componentDidMount() {
-    this.props.getTableItems(getURIQuery(window.location.href));
-  }
-
-  render() {
-    const {
-      getTableItems,
-      sortBy,
-      sortOrder,
-      error,
-      status,
-      results,
-    } = this.props;
+const ModelsTable = ({
+  getTableItems,
+  sortBy,
+  sortOrder,
+  error,
+  status,
+  results,
+}) => {
+  useEffect(() => {
```

Comment:
You can use `callOnMount` HOC instead of calling `useEffect` here for a cleaner look ( see ReactApp index.js for an example ).

Reply:
TBH that makes it more complex in my opinion, instead of seeing the effect as part of the component you need to look in another file.